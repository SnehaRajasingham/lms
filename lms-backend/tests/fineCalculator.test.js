const { calculateFine } = require('../utils/fineCalculator');

describe('Fine Calculator', () => {
  it('should return 0 if returned within 7 days', () => {
    const fine = calculateFine(new Date('2024-01-01'), new Date('2024-01-05'));
    expect(fine).toBe(0);
  });

  it('should calculate fine for overdue returns', () => {
    const fine = calculateFine(new Date('2024-01-01'), new Date('2024-01-10'));
    expect(fine).toBe(20);
  });

  it('should match fine calculation snapshot', () => {
    const fine = calculateFine(new Date('2024-01-01'), new Date('2024-01-10'));
    expect(fine).toMatchSnapshot();
  });
});
