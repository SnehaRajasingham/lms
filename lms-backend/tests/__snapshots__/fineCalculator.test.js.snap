// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fine Calculator should match fine calculation snapshot 1`] = `20`;
